### PR TITLE
refactor: separate propsStores from the initial value of props

### PR DESCRIPTION
**Pull request by divdavem**

### CODE REVIEW

File: core/src/types.ts [hunk 1]
```diff
@@ -4,16 +4,24 @@ export type ValuesOrReadableSignals<T extends object> = {
...
-	props?: ValuesOrWritableSignals<U>;
+	propsStores?: WritableSignals<U>;
+
+	/**
+	 * Object containing, for each property its initial value.
+	 * If a store is provided in {@link PropsConfig.propsStores|propsStores} for the same property,
+	 * the initial value provided in props is ignored.
+	 * When the value of a property is undefined or invalid, the value from the config is used.
+	 */
+	props?: Partial<U>;
```
Comment (MarkoOleksiyenko) [inline core/src/types.ts:24]: does it make sense to call it directly initialProps?

**Review by MarkoOleksiyenko — APPROVED** [review]
thanks for the PR @divdavem ! lgtm

**Comment by divdavem** [comment]
@MarkoOleksiyenko Thank you for your review!

**Comment by divdavem** [comment]
As discussed, I am closing this PR and I am opening [this one](https://github.com/AmadeusITGroup/AgnosUI/pull/332) instead.

**Pull request closed without merging**